JavaScript development/Frameworks/React.md

```diff
@@ -0,0 +1,536 @@
+  // ..
+};
+
+export default ComponentName;
+```
+
+Exporting stateless components:
+``` jsx
+// stateless components
+function PureComponent(/* props */) {
+  // ...
+}
+
+PureComponent.propTypes = {
+  // ...
+};
+
+export default PureComponent;
+
+// stateless components with no prop validation
+export default () => (/* ... */);
+```
+
+### Rendering Arrays of Data
+
+For iterating arrays:
+- use **map** method to render items
+- you should have a unique key for each element
+- **don't use array's index as a key**
+
+``` jsx
+render() {
+  // ...
+  return (
+    <ul>
+      {
+        booksArray.map((book) => {
+          <ListItem key={book.id} title={book.title} /> 
+        }); 
+      }
+    </ul>
+  );
+}
+```
+
+### Binding
+Don't bind functions inline, bind them in the class constructor.
+``` jsx
+// bad
+class ComponentName extends React.Component {
+  onClickHandler() {
+    // ...
+  }
+
+  render() {
+    return (
+      <ChildComponent onClick={this.onClickHandler.bind(this)} />
+    );
+  }
+}
+    
+// good
+class ComponentName extends React.Component {
+  constructor(args) {
+    super(args);
+
+    this.onClickHandler = this.onClickHandler.bind(this);
+  }
+
+  onClickHandler() {
+    // ...
+  }
+
+  render() {
+    return (
+      <ChildComponent onClick={this.onClickHandler} />
+    );
+  }
+}
+```
+
+### CSS Modules and Components
+Each component should have it's own stylesheet if CSS is needed for it.
+
+```
+...
+├─ Login
+⎮   └─ Login.js
+⎮   └─ Login.scss
+...
+```
+
+Importing:
+
+``` jsx
+// ...
+import styles from './ComponentName.scss';
+
+class ComponentName extends React.Component {
+  // ...
+  render() {
+    return (
+      <Heading className={styles.specialHeading} />
+    );
+  }
+}
+```
+
+If you need to pass more than one CSS class name to a component or want to use conditional classes, use the [`classnames`](https://www.npmjs.com/package/classnames) library.
+
+``` jsx
+import classNames from 'classnames';
+// ...
+import styles from './ComponentName.scss';
+
+class ComponentName extends React.Component {
+  // ...
+  render() {
+    return (
+      <Heading
+        className={classNames(styles.specialHeading, styles.largeHeading)}
+      />
+    );
+  }
+}
+```
+
+## React + MobX
+*For general information about MobX, checkout the official documentation:* [`mobx.js.org`](https://mobx.js.org/).
+
+## Decorators
+Use decorators wherever you can because it makes code more readable and declarative.
+
+### Component Declaration
+Wrap components that handle app's state with *observer* from the `mobx-react` library.
+
+``` jsx
+import {observer} from 'mobx-react';
+// ...
+@observer
+class ComponentName extends React.Component {
+  // ...
+}
+```
+
+### Local Component State
+If MobX is used for state management:
+- preffer using MobX *observable* instead of *this.state* for component's local state; don't use both
+- wrap local state in a single object just as it would be in the built in state
```

Comment:
Make it clear that some name other than `state` should be used. It might also make sense to define a name here, so we're using the same name everywhere. @andreicek